framework/mesh/UnpartitionedMesh/unpartmesh_00a_general.cc

```diff
@@ -90,6 +90,50 @@ void chi_mesh::UnpartitionedMesh::ComputeCentroidsAndCheckQuality()
     }   // if polyhedron
   }     // for cell in raw_cells
 
+  Chi::log.Log0Verbose1() << "Checking face sizes";
+  size_t cell_id = 0;
+  for (auto cell : raw_cells_)
+  {
+    if (cell->type == CellType::POLYGON)
+    {
+      size_t f = 0;
+      for (const auto& face : cell->faces)
+      {
+        const auto& v0 = vertices_.at(face.vertex_ids[0]);
+        const auto& v1 = vertices_.at(face.vertex_ids[1]);
+        ChiLogicalErrorIf((v1 - v0).Norm() < 1.0e-12,
+                          "Cell " + std::to_string(cell_id) + " (centroid=" +
+                            cell->centroid.PrintStr() + ") face " +
+                            std::to_string(f) + ": Face has length < 1.0e-12.");
+        ++f;
+      }
+    } // if polygon
+    if (cell->type == CellType::POLYHEDRON)
+    {
+      size_t f = 0;
+      for (const auto& face : cell->faces)
+      {
+        size_t num_face_verts = face.vertex_ids.size();
+        for (size_t s = 0; s < face.vertex_ids.size(); ++s)
+        {
+          size_t fvp1 = (s < (num_face_verts - 1)) ? s + 1 : 0;
+
+          const auto& v0 = vertices_.at(face.vertex_ids[s]);
+          const auto& v1 = vertices_.at(face.vertex_ids[fvp1]);
+
+          ChiLogicalErrorIf((v1 - v0).Norm() < 1.0e-12,
+                            "Cell " + std::to_string(cell_id) + " (centroid=" +
+                              cell->centroid.PrintStr() + ") face " +
+                              std::to_string(f) + " side " + std::to_string(s) +
+                              ": Side has length < 1.0e-12.");
+        }
+
+        ++f;
+      }
+    } // if polyhedron
+    ++cell_id;
+  } // for cell in raw_cells
+
   if (num_negative_volume_elements > 0)
     Chi::log.LogAllWarning()
       << "Cell quality checks detected " << num_negative_volume_elements
```

Comment:
can this error message be more explicit by saying negative volumes. this would help the user focus on clockwise/ccw issues in their mesh

Reply:
was my comment unhelpful ?